federation/pkg/kubefed/init/init.go

```diff
@@ -934,3 +939,55 @@ func updateKubeconfig(config util.AdminConfig, name, endpoint, kubeConfigPath st
 
 	return nil
 }
+
+func cleanupKubeconfig(config util.AdminConfig, name, kubeConfigPath string, dryRun bool) error {
+	po := config.PathOptions()
+	po.LoadingRules.ExplicitPath = kubeConfigPath
+	kubeconfig, err := po.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+
+	delete(kubeconfig.Clusters, name)
+	delete(kubeconfig.AuthInfos, name)
+	delete(kubeconfig.Contexts, name)
+
+	if !dryRun {
```

Comment:
Why bother passing dryRun to this function?  Why not just avoid calling it if dryRun is true?